### PR TITLE
CHK-482: Fix check for number field validity

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What is the purpose of this pull request?

Update the checks for the number validity to use `address.number.valid`.

#### What problem is this solving?

Fix the checks for the number validity, that where previously using the `address.number.value` instead of `address.number.valid`, which caused issues on countries that do not have number in the addresses.

#### How should this be manually tested?

[Workspace](https://lucas--vtexgame1geo.myvtex.com/checkout/cart/add?sku=312&qty=1&seller=1&sc=2).

You can check this works by using the "United Emirate Arabes" country in the workspace above and with the address "W Corniche Road" (you can select the ~first~ only option). You should be able to proceed to the payment step without issue.

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Review by wdsrocha — CHANGES_REQUESTED** [review]
LGTM, but the changelog is missing

**Comment by lucasecdb** [comment]
> LGTM, but the changelog is missing

done :-)

**Review by wdsrocha — DISMISSED** [review]

File: react/containers/DeliveryChannelsContainer.js [hunk 1]
```diff
@@ -123,13 +123,18 @@ class DeliveryChannelsContainer extends Component {
...
     const postalCode = this.props.rules.fields.find(
-      field => field.name === 'postalCode'
+      (field) => field.name === 'postalCode'
     )
 
     if (
       (postalCode &&
         postalCode.postalCodeAPI &&
         !nextProps.address.postalCode.valid) ||
+      // For unnmapped countries, we don't want to update
```
Comment (darlenedms) [inline react/containers/DeliveryChannelsContainer.js:133]:
```suggestion
      // For unmapped countries, we don't want to update
```
😝

**Comment by caetano1** [comment]
Hey, @lucasecdb ! I've tested and tried to finish the purchase using "boleto bancário", but after clicking on "buy", VTEX ID kept popping up to me, and repopping after authentication using key sent to email. Is it related to this PR?

**Comment by lucasecdb** [comment]
> Hey, @lucasecdb ! I've tested and tried to finish the purchase using "boleto bancário", but after clicking on "buy", VTEX ID kept popping up to me, and repopping after authentication using key sent to email. Is it related to this PR?

I don't think so. VTEX ID is sometimes shown when you're using a second purchase account during the place order, according to [this article](https://help.vtex.com/faq/por-que-foi-solicitado-login-no-checkout-finalizacao-de-compra--frequentlyAskedQuestions_2705).

**Review by alvim — APPROVED** [review]

**Review by wdsrocha — APPROVED** [review]

**Comment by caetano1** [comment]
> > Hey, @lucasecdb ! I've tested and tried to finish the purchase using "boleto bancário", but after clicking on "buy", VTEX ID kept popping up to me, and repopping after authentication using key sent to email. Is it related to this PR?
> 
> I don't think so. VTEX ID is sometimes shown when you're using a second purchase account during the place order, according to [this article](https://help.vtex.com/faq/por-que-foi-solicitado-login-no-checkout-finalizacao-de-compra--frequentlyAskedQuestions_2705).

Hmm got it! Thank you!

**Review by jeffersontuc — APPROVED** [review]

**Comment by jeffersontuc** [comment]
I just saw it needs rebase

**Review by darlenedms — APPROVED** [review]

**Review by wdsrocha — APPROVED** [review]

**Pull request merged**